tests/unit/resources/test_params.py

```diff
@@ -88,14 +88,25 @@ def test_service_action_params_constants(self):
         self.assertEqual(params['Param3'], True,
             'Parameter not set from boolean constant')
 
+    def test_service_action_params_input(self):
```

Comment:
Is there a test somewhere that checks when the source is input and the user provides the param that the param is in the returned dict?